plugin-gradle/src/main/java/com/diffplug/gradle/spotless/SpotlessTask.java

```diff
@@ -303,9 +304,21 @@ private void check(Formatter formatter, List<File> outOfDate) throws Exception {
 	/** Returns an exception which indicates problem files nicely. */
 	GradleException formatViolationsFor(Formatter formatter, List<File> problemFiles) {
 		return new GradleException(DiffMessageFormatter.builder()
-				.runToFix("Run 'gradlew spotlessApply' to fix these violations.")
+				.runToFix("Run '" + calculateGradleCommand() + " " + getTaskPathPrefix() + "spotlessApply' to fix these violations.")
 				.formatter(formatter)
 				.problemFiles(problemFiles)
 				.getMessage());
 	}
+
+	private String calculateGradleCommand() {
+		return Os.isFamily(Os.FAMILY_WINDOWS)
+				? "gradlew.bat"
+				: "./gradlew";
+	}
+
+	private String getTaskPathPrefix() {
+		return getProject().getPath().equals(":")
+				? ":"
+				: getProject().getPath() + ":";
```

Comment:
If we're going to take responsibility for an optimal error message, should the message for single-project build be `spotlessApply` rather than `:spotlessApply`?

Reply:
I would keep the `:` as it is strictly speaking more exact.

Reply:
Okay.